### DISCUSSION TITLE
Handheld Itemfinder works in ruins/wreck when the description suggests it only works within your submarine.

### DISCUSSION THREAD

**Discussion by Crazyjr20** [FakeFishGames/Barotrauma#14209] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The assistant talent item **"Handheld Itemfinder"** is described as "A small handheld device with a screen, capable of easily finding items within the submarine."
However, it also works within ruins and wrecks and supposedly any other .sub structures, not just the player submarine. 
This allows you to easily locate any item within them as well as get a clear list of everything there will be.
I assume this is not intentional and rather an oversight, so I'm posting it as a bug report as I didn't see any other reports about it so I am unsure if you devs were aware of it, though I am a bit unfamiliar with github so I may of missed one.
![20240625233306_1](https://github.com/FakeFishGames/Barotrauma/assets/18705965/6d203613-1bf5-406a-93e2-d2aa3fce24a8)
![20240625234021_1](https://github.com/FakeFishGames/Barotrauma/assets/18705965/5b8a6f8d-a71f-45a5-8781-380509addefc)

### Reproduction steps

1. Start up a ruin mission or wreck mission.
2. Spawn in "Handheld Itemfinder".
3 Teleport to the destination.
4. Type into the dropdown box to get a list of all items.
5. Click the item you want to locate.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.8.0 (Summer Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
https://www.youtube.com/watch?v=4sQ4kSHDYxs&t=244s

**Comment by Jasontti** [comment]
Tested a fix for this.
Item finders now only work in their owners sub.
Closing discussion as completed.
